docs/transaction-api.asciidoc

```diff
@@ -64,13 +64,51 @@ Arguments:
 
 * `options` - The following options are supported:
 
+** `blocking` - Blocks the associated transaction from ending till this span is ended. Blocked spans
+    automatically internally creates a task. Defaults to false
+
 ** `parentId` - Parent id associated with the new span. Defaults to current transaction id
 
-** `sync` - Denotes if the span is blocking (sync) or non-blocking(async). 
+** `sync` - Denotes if the span is synchronous or asynchronous.
+
+When a span is started it will measure the time until <<span-end,`span.end()`>> is called. Blocked
+spans allow users to control the early closing of <<custom-managed-transactions, managed transactions>> in some cases.
+
+See also the <<span-api,Span API>> docs for details on how to use custom spans.
+
+[float]
+[[transaction-block]]
+==== `transaction.block(flag)`
+
+[source,js]
+----
+transaction.block(true)
+----
+
+Blocks the currently running auto-instrumented or custom transactions from getting automatically closed once the associated spans are ended.
 
-When a span is started it will measure the time until <<span-end,`span.end()`>> is called.
+Arguments:
+
+* `flag` - Boolean. Defaults to false.
+
+Most of the time, users shouldn't need to block the running transaction as the agent tries to account for all of the async network activity and capture
+the required information. The block API is useful when the user feels the auto instrumented code missed some important activity and ends up prematurely
+closing the transaction and the page contains lots of asynchronous tasks that cannot be automatically instrumented (eg: web workers).
```

Comment:
This is a bit of a run-on sentence.
```
The block API is useful when the user feels the auto instrumented code missed some important activity and ends
up prematurely closing the transaction and the page contains lots of asynchronous tasks that cannot be
automatically instrumented (eg: web workers).
```

What about this instead?
```
The block API is useful when you think auto instrumented code misses important activity and prematurely ends a transaction.
An example of this might be a page that contains lots of asynchronous tasks that cannot be automatically instrumented, like web workers.
```